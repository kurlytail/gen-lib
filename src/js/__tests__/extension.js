import FS from 'fs';
import Extension from '../extension';

jest.mock('fs');

const FIXTURES = {
    GENERATOR: { design: 'b', map: 'c', options: 'd', labels: [] },
    TEMPLATEDESCRIPTION: {
        context: 'test'
    }
};

describe('# Extension', () => {
    beforeEach(() => {
        FS.readFileSync.mockReset();
    });
    describe('## constructor', () => {
        it('### should construct default object', () => {
            const extension = new Extension('a', FIXTURES.GENERATOR);
            expect(extension).toMatchSnapshot();
        });
    });

    describe('## load', () => {
        it('### should load template from file', () => {
            FS.readFileSync.mockReturnValue('test template');
            const extension = new Extension('a', FIXTURES.GENERATOR);
            extension.load();
            expect(extension).toMatchSnapshot();
        });
    });

    describe('## template', () => {
        it('### should generate extension', () => {
            FS.readFileSync.mockReturnValue('test template <%labels.forEach(lab => {%>"<%=lab%>"<%})%> <%=context%>');
            const extension = new Extension('a', FIXTURES.GENERATOR);
            extension.load();

            let text = extension.generate('test', FIXTURES.TEMPLATEDESCRIPTION);
            expect(text).toMatchSnapshot();

            text = extension.generate(undefined, FIXTURES.TEMPLATEDESCRIPTION);
            expect(text).toMatchSnapshot();

            text = extension.generate(['test', 'test2'], FIXTURES.TEMPLATEDESCRIPTION);
            expect(text).toMatchSnapshot();
        });
    });
});
