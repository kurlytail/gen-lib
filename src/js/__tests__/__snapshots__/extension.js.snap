// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`# Extension ## constructor ### should construct default object 1`] = `
Extension {
  "_file": "a",
  "_generator": Object {
    "design": "b",
    "labels": Array [],
    "map": "c",
    "options": "d",
  },
}
`;

exports[`# Extension ## load ### should load template from file 1`] = `
Extension {
  "_file": "a",
  "_generator": Object {
    "design": "b",
    "labels": Array [],
    "map": "c",
    "options": "d",
  },
  "_template": "test template",
}
`;

exports[`# Extension ## template ### should generate extension 1`] = `"test template \\"test\\""`;

exports[`# Extension ## template ### should generate extension 2`] = `"test template "`;

exports[`# Extension ## template ### should generate extension 3`] = `"test template \\"test\\"\\"test2\\""`;
